### ISSUE TITLE
[Not a bug] Is it possible to use Magisk with default Google Android Emulator with exactly Android 9.0 image

### ISSUE THREAD

**Issue by AndNovAtor** [topjohnwu/Magisk#7069]
I tried Magisk with multiple images on Google Android Emulator (with Android Stuido), and only with Android 9.0 images I have proplems. I can't "install" (by ramdisk) Magisk with this exactly Android version. For example, with Android 8.0 and Android 11.0 Google images it works, simple install by patch ramdisk (in-app patch also works).

I tried multiple Magisk version, I even tried MagiskDelta - no success. I tried multiple Android images - Google APIs x86, x86_x64, Google Play x86 - no success.

I use shakalaca's [MagiskOnEmulator](https://github.com/shakalaca/MagiskOnEmulator) `patch.bat`. I use Windows 10 x64. For that, I copy `ramdisk.img` from, for example, `<sdk path>/system-images/android-28/google_apis/x86/ramdisk.img`, to MagiskOnEmulator folder (Magisk app also copied there and renamed as `magisk.zip`).

I tried 2 ways - simple patch and in-app boot.img patch.

1) When I simply run "patch.bat", it sucessfully patched ramdisk.img, but when I copy patched file to AVD folder and start AVD, it doesn't have any effect, Magisk app shows "Ramdisk: No", so it's not working

2) When I try "patch.bat manager" to patch boot.img in-app, it doesn't work in the app, with _"any"_ android 9.0 image it failed with
`! Unsupported/Unknown image format`
`! Installation failed`

3) When I replace magisk.zip with MagiskDelta, in-app patch as from 2) suddenly works, but still, as for 1), it again doesn't have any effect, Magisk was not loaded....

So, is it something "wrong" with Google Android 9.0 ramdisk? Or is there some workaround?
Or I don't know something?

(I can use Magisk with another Android Emulator's Android image versions, so it isn't a problem, but still, I'm curious.
Sorry for possible duplicate)

**Comment by MagiskBot** [comment]
Invalid bug report, automatically closed.
Please report issues using the latest debug Magisk build (version code: 26102).

**Comment by yujincheng08** [comment]
https://github.com/topjohnwu/Magisk/blob/master/scripts/avd_patch.sh

**Comment by AndNovAtor** [comment]
> https://github.com/topjohnwu/Magisk/blob/master/scripts/avd_patch.sh

Well... I tried build from source (git clone from current master branch with submodules and "symlinks" config) and patch `ramdisk.img` with `build.py` - STILL doesn't work. And _exactly_ for Google Android 9.0 image from Android SDK.

I tried both - simply put patched ramdisk to avd folder, or patch default sdk ramdisk (system-images\android-28\google_apis\x86). In both situations Magisk app show "Ramdisk: No".

Again, this happens ONLY with 9.0 image of Googe Android Emulator, with all another versions Magisk works like a charm.
